web/yo/app/scripts/controllers/gene.js

```diff
@@ -1522,52 +1521,172 @@ angular.module('oncokbApp')
+                                mapPath = drug + '/' + geneName + '/' + mutationUuid + '/cancerTypes/' + cancerTypeUuid + '/' + therapyUuid + '/name';
+                                firebaseConnector.setMap(mapPath, content);
+                            });
+                        });
+                        break;
+                }
+            }
+
+            function removeMapWithoutValidatingStatus (geneName, mutationUuid, cancerTypeUuid, therapyUuid, therapyUuids){
+                var mapPath = '';
+                _.each(therapyUuids, function (drug) {
+                    mapPath = drug + '/' + geneName + '/' + mutationUuid + '/cancerTypes/' + cancerTypeUuid + '/' + therapyUuid;
+                    firebaseConnector.removeMap(mapPath).then(function(){
+                        firebaseConnector.ref('Map/' + drug + '/' + geneName + '/' + mutationUuid + '/cancerTypes').once('value', function (doc) {
+                            if (!doc.val()) {
+                                mapPath = drug + '/' + geneName + '/' + mutationUuid + '/mutationName';
+                                firebaseConnector.removeMap(mapPath);
+                            }
+                        })
+                    })
+                });
+            }
+
+            function changeTherapyStatusInMap(dataType, geneName, mutationUuid, cancerTypeUuid, therapyUuid, therapyUuids) {
                 switch (dataType) {
+                    case 'mutation':
+                        break;
+                    case 'tumor':
+                        break;
                     case 'treatment':
-                        var TherapyUuids = _.flatten(mainUtils.therapyStrToArr(treatment.name));
+                        _.each(therapyUuids, function (drug) {
+                            var mapPath = drug + '/' + geneName + '/' + mutationUuid + '/cancerTypes/' + cancerTypeUuid + '/' + therapyUuid + '/status';
+                            firebaseConnector.setMap(mapPath, 'reviewed');
+                        });
                         break;
-                };
-                switch (type) {
-                    case 'name':
-                        switch (dataType) {
-                            case 'mutation':
-                                break;
-                            case 'tumor':
-                                break;
-                            case 'treatment':
-                                var oldTherapyUuids = _.flatten(mainUtils.therapyStrToArr(oldContent));
-                                changeMap('remove', treatment.name_uuid, mapPath, oldTherapyUuids);
-                                changeMap('save', treatment.name_uuid, mapPath, TherapyUuids, treatment.name);
-                                break;
+                }
+            }
+
+            function updateDrugMapWhenAcceptDeletionOrRejectAddition(dataType, mutation, tumor, treatment, mapInfo){
+                switch (dataType) {
```

Comment:
Technically, all switch should attach a default handler, and leave a comment there for all scenarios that not qualify previous cases.